### PR TITLE
backend/s3: remove use_legacy_workflow argument

### PR DESCRIPTION
**Pull request by jar-b**
In Terraform 1.8 the `use_legacy_workflow` argument will be removed to encourage consistency with the AWS SDKs. The backend will now search for credentials in the same order as the default provider chain in the AWS SDKs and AWS CLI.

```console
% TF_ACC=1 go test ./internal/backend/remote-state/s3/...
ok      github.com/hashicorp/terraform/internal/backend/remote-state/s3 142.398s
```

Relates #30443 
Relates #34337

## Target Release

1.8.x

### UPGRADE NOTES 
 

* backend/s3: The `use_legacy_workflow` argument has been removed to encourage consistency with the AWS SDKs. The backend will now search for credentials in the same order as the default provider chain in the AWS SDKs and AWS CLI.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ UPGRADE NOTES:
 
 * The first plan after upgrading may show resource updates with no apparent changes if `-refresh-only` or `-refresh=false` is used. The fix introduced for #34567 may require rewriting the state for some resources, which will be done automatically during the first normal plan and apply operation.
 
+* backend/s3: The `use_legacy_workflow` argument has been removed to encourage consistency with the AWS SDKs. The backend will now search for credentials in the same order as the default provider chain in the AWS SDKs and AWS CLI.
```
Comment (ewbankkit) [inline CHANGELOG.md:7]: Breaking changes (which I guess this is as any configuration that contains `use_legacy_workflow` will no longer work) are not called out separately?

File: internal/backend/remote-state/s3/backend.go [hunk 2]
```diff
@@ -1026,14 +1010,6 @@ func (b *Backend) Configure(obj cty.Value) tfdiags.Diagnostics {
 		Token:                   stringAttr(obj, "token"),
 	}
 
-	// The "legacy" authentication workflow used in aws-sdk-go-base V1 will be
-	// gradually phased out over several Terraform minor versions:
-	//
-	// 1.6 - Default to `true` (prefer existing behavior, "opt-out" for new behavior)
-	// 1.7 - Default to `false` (prefer new behavior, "opt-in" for legacy behavior)
-	// 1.8 - Remove argument, legacy workflow no longer supported
-	cfg.UseLegacyWorkflow = boolAttr(obj, "use_legacy_workflow")
```
Comment (ewbankkit) [inline internal/backend/remote-state/s3/backend.go:1035]: Can we now remove `UseLegacyWorkflow` from https://github.com/hashicorp/aws-sdk-go-base?

**Review by ewbankkit — APPROVED** [review]
A couple of comments, otherwise LGTM :rocket:.

File: internal/backend/remote-state/s3/backend.go (hunk 2 above)
Comment (jar-b) [reply internal/backend/remote-state/s3/backend.go:1035]: I believe so as I don't think we ever exposed it in the AWS provider. I'll open an issue.

File: CHANGELOG.md (hunk 1 above)
Comment (jar-b) [reply CHANGELOG.md:7]: Discussed separately with the core team, and the recommendation was to leave in Upgrade Notes with a link to the upgrade guide once it is fully updated for Terraform 1.8.

File: internal/backend/remote-state/s3/backend.go (hunk 2 above)
Comment (jar-b) [reply internal/backend/remote-state/s3/backend.go:1035]: Opened https://github.com/hashicorp/aws-sdk-go-base/issues/946

**Pull request merged**